src/main/java/org/dependencytrack/resources/v1/ProjectResource.java

```diff
@@ -497,6 +499,44 @@ public Response deleteProject(
         }
     }
 
+    @POST
+    @Path("/batchDelete")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(
+            value = "Deletes a list of projects specified by their UUIDs",
+            code = 204
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 401, message = "Unauthorized"),
+            @ApiResponse(code = 403, message = "Access to one or more of the specified projects is forbidden")
+    })
+    @PermissionRequired(Permissions.Constants.PORTFOLIO_MANAGEMENT)
+    public Response deleteProjects(
+            @ApiParam(value = "A list of one or more project UUIDs", required = true)
+            List<UUID> uuids) {
+        List<UUID> inaccessibleProjects = new ArrayList<>();
+        try (QueryManager qm = new QueryManager()) {
+            for (UUID uuid : uuids) {
+                final Project project = qm.getObjectByUuid(Project.class, uuid, Project.FetchGroup.ALL.name());
+                if (project != null) {
+                    if (qm.hasAccess(super.getPrincipal(), project)) {
+                        LOGGER.info("Project " + project + " deletion request by " + super.getPrincipal().getName());
+                        qm.recursivelyDelete(project, true);
+                    } else {
+                        inaccessibleProjects.add(uuid);
+                    }
+                } else {
+                    LOGGER.warn("No project found by UUID " + uuid);
+                }
+            }
+            if (!inaccessibleProjects.isEmpty()) {
```

Comment:
How about a 207 (Multi-Status) instead of a 204? That maps semantically better to a mixed outcome when some (but not all) projects are inaccessible. I'm thinking something like this:
```
            if (!inaccessibleProjects.isEmpty()) {
                // not all are inaccessible
                if (inaccessibleProjects != uuids) { 
                    return Response.status(207)
                            .entity(inaccessibleProjects)
                            .build();
                }
                // all are inaccessible
                return Response.status(Response.Status.FORBIDDEN)
                        .entity(inaccessibleProjects)
                        .build();
            }
```